pkg/gui/menu_panel.go

```diff
@@ -30,9 +31,25 @@ func (gui *Gui) createMenu(opts types.CreateMenuOptions) error {
 		})
 	}
 
+	maxColumnSize := 1
+
+	for _, item := range opts.Items {
+		if item.LabelColumns == nil {
+			item.LabelColumns = []string{item.Label}
+		}
+
+		if item.OpensMenu {
+			item.LabelColumns[0] = presentation.OpensMenuStyle(item.LabelColumns[0])
+		}
+
+		maxColumnSize = utils.Max(maxColumnSize, len(item.LabelColumns))
+	}
+
 	for _, item := range opts.Items {
-		if item.OpensMenu && item.LabelColumns != nil {
-			return errors.New("Message for the developer of this app: you've set opensMenu with displaystrings on the menu panel. Bad developer!. Apologies, user")
+		if len(item.LabelColumns) < maxColumnSize {
+			// we require that each item has the same number of columns so we're padding out with blank strings
+			// if this item has too few
```

Comment:
I considered making the downstream code more lenient i.e. just dealing better with being passed rows of different column numbers, but I think it's better to be explicit at the call site, given how easily we could accidentally forget to add a column to a given row when rendering, say, the commits panel.